impl: Customer Order History Page

kindubadhan opened the pull request:
This branch implements an order history page for customers, so that customers can see their previous orders and what items they contained. It shows a summary of each order in a table format, with more detailed information about each order and the included items in sections below the table.

The following files were changed:
- App.js: Import and create the route
- header.js to import the order button in header section under the login section.
- orderAction.js to get the order 
- orderListScreen.js to implement the order history code including the format of that page.

GitHub issue #72

kindubadhan committed:
impl: Add Order List Page
In the frontend folder:
- Added orderList.js page to show order history of the customer.
- Added orderConstant.js, orderReducer.js, and orderAction.js file in src folder.

Related to Github issue #72.

kindubadhan committed:
impl: Update The OrderListScreen Page
In the Frontend Folder,
- Update the orderAction file, in order to implement the order history page.
- Update the OrderListScreen file using retrieve order function to get customer's all order history.
- Created a Table in this page with orderId, customer name, total price, and total number of items.
- Also added a button named Detail,
  which will take customer to that particular order that user wants to see using order ID.
- In order detail section, provide Shipping information using
  customer name, address, total price and order Items.
- In order items, customer can see all items that they purchased
  with item images, title, price and no. of quantities.

  Github issue #72

kindubadhan committed:
Fix: Fix The Total Price of each Order
- Multiply amount with .01 to make exact total price where unit is cents.
- Also used {toFixed(2)} code to show only 2 digits after decimal.

Github issue #72

yu-zha reviewed (changes requested):
Please check the package-lock.json file.
It seems to upload too much stuff which is not used.
You may copy the package-lock.json file and overwrite it from the main branch.

kindubadhan committed:
Fix: Package-lock.json

kindubadhan committed:
Updated Package.json file

SuperRedTrain reviewed (changes requested):
This pull request is appropriately labelled and has the issue linked. But still have some issues:
(1) The title should be something like "impl: Chang X in Function Y"
(2) The first comment should give an accurate and detailed summary of the changes in PR. It also needs to explain why the changes were made.
(3) This pull request implements display-customer-order-history function, I think it do not need to contain commits of implementing place order function.
(4) Are codes related to manage Orders_List state by redux still used to display order history? If not, could you please delete them.

kindubadhan committed:
Fix: Package-lock.json file
In Frontend folder
- Fix Package-lock.json file because it was overwritten and size became huge.
- Also Fix package.json file because it has mdbreact package
  which we actually did not use at all. so delete that package too from package.json.
- Also make some changes in orderListScreen.
  It has some extra lines and imported some other dependecies,
  which one also did not use in that file.

  Github issue #72

kindubadhan committed:
Fix: Deleted Some Redux Files
In Frontend Folder
- Delete orderConstant.js and orderReducers.js files.
  Because these files were created to show order History using Redux.
  But later, we did not use Redux at all.
  So these files have no use at all in this program so deleted them.
- Also delete some code from store.js
  which is connected to redux files for order history.
- Deleted some extra codes and lines from orderListScreen file which has no use any more.

  Github issue #72.

London-Alex committed:
Merge branch 'main' into kulwinder-alex-customer-order-history

SuperRedTrain reviewed (approved):
I have tested on my local machine. The customer order history page works well. Thank you for your efforts, Kulwinder and Alex!
